### ISSUE TITLE
Win 10 No Connection to Bluetooth

### ISSUE THREAD

**Issue by LividJava** [Ryochan7/DS4Windows#631]
This has become somewhat infuriating, I have paired my controller to my PC and DS4W has seen my controller just fine in the past. But connecting to it now provides no dice. I can connect fine with Micro-B, but bluetooth for whatever reason just wont connect. I have unpaired and re-paired my controller and even reset the controller. I specifically switched to this fork hoping it would resolve the issue. to no avail. If you need any more information i can provide and will edit this one as it comes up.

Windows sees the controller and connects to it (Wireless Controller) just fine, but DS4W isn't latching on to it for whatever reason.

**Comment by mika-n** [comment]
Which version of DS4Win application?
What do you see in  DS4Windows Logs option screen?
Are you using official Sony DS4 dualshock controller or some 3rd party "compatible" version?

**Comment by Ryochan7** [comment]
Is the controller an official Sony brand DS4 or third party?
Does the Windows Game Controller settings window (accessed from Devices and Printers in Windows Control Panel) see the controller and register inputs?
Are you using HidGuardian installed from another program but you are running an older version of DS4Windows?

Making sure that Windows can see the controller and register inputs is the first step. If it does not work in Windows without running DS4Windows then it won't work in DS4Windows. Then you can focus on potential issues with DS4Windows.

**Comment by LividJava** [comment]
it appears to be working now, but given how its functioned in the past i don't trust it. I'll close this issue until i can replicate/reproduce the issue. For reference i am using this dongle "NS-PCY5BMA2". I will open again in the future provided i have the information requested.

**Comment by v-fox** [comment]
This has been happening for me periodically. I suppose issues #675 and #711 are the same.
My DS4v2 would just flash in default colour and then eaither disconnect or stay without registering as input device. Weirdest part is that it was happening the same way in Linux... I even thought that it or BT dongle was dying. Reinserting dongle and/or reinstalling drivers for it and the controller somehow randomly helped but today it happened after ridiculous shenanigans of Windows 10 upgrade to version 1903 on the system running from M.2 NVMe on MB without BIOS/UEFI support for booting from it. I had to use a chain of 3 primary bootloaders on HDD partition (GRUB2-UEFI->rEFInd->Clover->W10-UEFI_on_NVMe_NTFS_W10_partition_instead_of_EFI->W10-kernel) and then manually fix UI that broke after "successful" update by `Dism /Online /Cleanup-Image /RestoreHealth /Source:wim:<wim-file from installer>:1`. But BT in DS4 also broke after all that.

After hours of throwing crap into a wall in hopes that it would stick I found out that even though DS4 were pairing with dongle, it was not getting HID profile from Windows. And if I force-enable it via its entry in "Devices and printers" that would lead to "Conflicting Address Range: The specified address range conflicts with the address space." failure in "Bluetooth HID device". I also noticed that BT services are in "manual" launch mode and attempts to in changing it fail with "wrong parameter" error. So, I force-uninstalled driver entry and then [used regedit to force automatic BT service launch](https://answers.microsoft.com/en-us/windows/forum/all/changing-bluetooth-user-support-service-from/f453e44c-4215-4a1e-b987-f82b356a0dae?auth=1&page=2), then rebooted. Although "Bluetooth user service" didn't launch at all, BT HID started working and DS4Windows has acknowledged DS4.

Isn't Bluetooth just a wonderfull thing ? Especially, considering that Linux BT stack sometimes fails in the same exact manner where HID support on paired device has to be force-enabled somehow.
